### ISSUE TITLE
Default nuxt error page bleeds CSS styles

### ISSUE THREAD

**Issue by Aareksio** [nuxt/ui-templates#91]
### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.13.2`
- Nuxt Version:     `3.0.0-27485997.c426f3e`
- Package Manager:  `npm@8.5.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

Repo:
https://github.com/Aareksio/nuxt3-issues/tree/error-style-bleed

CodeSandbox:
https://codesandbox.io/s/eloquent-cartwright-gkr2ll?file=/app.vue

TL;DR:
```html
<div>
  <nuxt-page/>
  <nuxt-link to="/404">404</nuxt-link>
</div>
```

https://user-images.githubusercontent.com/11166786/161754398-6a1bf886-01db-4332-a1fe-4559825689cb.mp4

### Describe the bug

After visiting default 404 page, global style is applied

**Comment by atinux** [comment]
I can confirm the issue, how could we workaround for this? Would switching to UnoCSS give us the ability to scope? @antfu @pi0

**Comment by antfu** [comment]
I could give it a shot
